### PR TITLE
Blaze: Campaign Detail Page: introduce campaign id into navigation

### PR DESCRIPTION
**Pull request by zwarm**
Parent #18532

This PR introduces the passing of "campaignId" from card tap to CampaignDetailFragment
Re:  `campaign.campaignID.toInt()` -  this will be updated as part of the FluxC updates for stats response.

**Note**: To keep the PR reviews small and manageable, I'll be breaking the CampaignDetail into separate PRs. This PR does not contain any logic to show the actual campaign detail view

**To test:**
- Install and launch the app
- Select a site that has blaze campaigns
- Navigate to My Site
- Tap on the Campaign Detail within the Blaze Card
- ✅ Verify the Campaign Detail view is launched

## Regression Notes
1. Potential unintended areas of impact
The campaign detail view doesn't launch

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual test

3. What automated tests I added (or what prevented me from doing so)
Updated `BlazeCardViewModelSliceTest` and `BlazeCardBuilderTest`

PR submission checklist:
- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist: N/A

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCard.kt [hunk 1]
```diff
@@ -53,7 +53,7 @@ fun BlazeCampaignsCard(
                 Column(
                     modifier = Modifier
                         .padding(start = 16.dp, end = 16.dp, bottom = 8.dp)
-                        .clickable { blazeCampaignCardModel.campaign.onClick() },
+                        .clickable { blazeCampaignCardModel.campaign.onClick(blazeCampaignCardModel.campaign.id.toInt()) },
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCard.kt]: Note: The `toInt()` will be replaced in a separate PR

**Review by AjeshRPai — APPROVED** [review]
Everything LGTM 👍🏼
